rust/cubestore/cubestore/src/sql/parser.rs

```diff
@@ -256,18 +282,20 @@ impl<'a> CubeStoreParser<'a> {
     pub fn parse_with_index(
         &mut self,
         table_name: ObjectName,
+        is_aggregate: bool,
     ) -> Result<SQLStatement, ParserError> {
         let index_name = self.parser.parse_object_name()?;
         self.parser.expect_token(&Token::LParen)?;
         let columns = self
             .parser
             .parse_comma_separated(Parser::parse_order_by_expr)?;
         self.parser.expect_token(&Token::RParen)?;
+        //TODO I use unique flag for aggregate index for reusing CreateIndex struct. When adding another type of index, we will need to parse it into a custom structure
```

Comment:
We can't  add a separate flag, because it is structure from sql-parser.rs, but we can write our own structure and write our own parser flow for indexes. Maybe you're right and I should do it.